File: regexlib/2021-5-15/python_re_test_file/regexlib_8008.py
# 8008
# (\[[Ii][Mm][Gg]\])(\S+?)(\[\/[Ii][Mm][Gg]\])
# POLYNOMIAL
# nums:4
# POLYNOMIAL AttackString:""+"[IMG]"*5000+"@1 _SLQ_2"

import re
from time import perf_counter

regex = """(\[[Ii][Mm][Gg]\])(\S+?)(\[\/[Ii][Mm][Gg]\])"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "[IMG]" * i * 10000 + "@1 _SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")

File: regexlib/2021-5-15/javascript_test_file/regexlib_8008.js
/* 8008
 * (\[[Ii][Mm][Gg]\])(\S+?)(\[\/[Ii][Mm][Gg]\])
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:""+"[IMG]"*5000+"@1 _SLQ_2"
 */
var REGEX = new RegExp("(\[[Ii][Mm][Gg]\])(\S+?)(\[\/[Ii][Mm][Gg]\])")
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '[IMG]'.repeat(i*10000) + '@1 _SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}